Create code to make and index connections

JoelEldo opened the pull request:
Code that connects power lines and power plants. Not complete yet. I still need to change it to accept custom names and connection information. I also need to convert some sections into functions. I am currently indexing the distance between the end of the line and power plant, as well as the ids of the power line and power plant. I plan on adding the location of the endpoint and power plant as well.

JoelEldo committed:
Create code to make and index connections

Code that connects power lines and power plants. Not complete yet. I still need to change it to accept custom names and connection information. I also need to convert some sections into functions.

JoelEldo committed:
Mapping for connections index

bytemedwb commented on connect_plants_lines.py:
```diff
@@ -0,0 +1,129 @@
+
+
+def prepare_elasticsearch():
+    mapping = json.loads(open(mappingFile).read())  # pulls mapping from a file
```
Minor nit. Don't go through the overhead of loading the mapping from the json file unless you are going to need it.  Move this to inside the if statement.

bytemedwb commented on connect_plants_lines.py:
```diff
@@ -0,0 +1,129 @@
+
+def calc_distance(long1, lat1, long2,
+                  lat2):  # calculates distance between two lat/long points using the Haversine formula
+    radius = 6371 * (10 ** 3)  # radius of the earth in meters
```
Document where this constant comes from.  And generally constants are all caps.
Make sure this value corresponds to the WGS84 Datum for the ellipsoid of the earth.

bytemedwb commented on connect_plants_lines.py:
```diff
@@ -0,0 +1,129 @@
+
+def get_pline_data():
+    print('starting load')
+    return geojson.loads(requests.get(data_url, allow_redirects=True).content)[
```
Why are we loading this data from the original URL and not from the ES database?

bytemedwb commented on connect_plants_lines.py:
```diff
@@ -0,0 +1,129 @@
+
+loaded_data = get_pline_data()
+elastic_connection = prepare_elasticsearch()
+print('retrieved and loaded')
```
Generally, Debug and status output like this should be more descriptive.  What was retrieved and loaded?

bytemedwb commented on connect_plants_lines.py:
```diff
@@ -0,0 +1,129 @@
+        plant_id = results['hits']['hits'][0]['_source']['gppd_idnr']
+        plant_location = results['hits']['hits'][0]['_source']['location']
+        distance1 = calc_distance(plant_location[0], plant_location[1], point[0], point[1])
+        DISTANCES.append({'line_id': line_id, 'plant_id': plant_id, 'distance': distance1})   #global distances dictionary
```
What is the purpose of the global DISTANCES dictionary.   My python pho is weak, but you could just return the distance as well.

bytemedwb commented on connect_plants_lines.py:
```diff
@@ -0,0 +1,129 @@
+    '_ID': position,
+    '_source': connection} for position, connection in enumerate(DISTANCES)
+)
+
```
I think I see now where the global DISTANCES is coming in.   I think this can be simplified to not use a global and just return the completed dictionary.  Consider if the json list can be built directly as the data is being processed to avoid the intermediate dictionary.

JoelEldo committed:
Update connection code

Removed useless imports. Changed, removed, and added some global variables. Added user input. Pulled power line data from elasticsearch instead of from the internet. Got rid of intermediate dictionary. Got rid of debugging print statement.

JoelEldo committed:
Removed redundant variable

Removed a redundant radius variable that I missed during first update.

JoelEldo committed:
Rename to lineplantdistances

The file finds the shortest distance from line endpoints to power stations, so this name is more accurate. Changed the index and some data structure names to be more accurate as well.

JoelEldo committed:
Adds file to connect power plants and lines 

Connects plants and lines within 500m of each other

JoelEldo committed:
Rename file 

Rename the file to make the format of the name match the format of the line_plant_distances file.

JoelEldo committed:
Rename line_plant_connections.py to plant_line_connections.py

Switched plant and line to better fit plant_line_distances.py

JoelEldo committed:
Update README

Update README with instructions.

JoelEldo committed:
Update README formatting

Updated the formatting of the readme.

JoelEldo committed:
Update readme information

Updated some outdated information in the readme

JoelEldo committed:
Update plant information pull

The ['_doc_source'] was a remnant from an earlier version of the code that I failed to remove.

JoelEldo committed:
Use correct endpoint

Code used to use the same end of the power line twice instead of using the opposite end. This caused the duplicate issue that Mr. Fraser noticed.

dclemenzi committed:
Merge branch 'master' into JoelEldo-patch-3
